### PR TITLE
Add support for HPyLong_(AsVoidPtr, AsDouble) & HPyErr_WriteUnraisable

### PR DESCRIPTION
**Pull request by qunaibit**
Required for porting `matplotlib` and `kiwi`

### CODE REVIEW

File: test/test_hpyerr.py [hunk 1]
```diff
@@ -662,3 +662,23 @@ def do_not_raise_exception(*args):
...
+            @INIT
+        """)
+        if not SUPPORTS_SYS_EXECUTABLE:
+            # if sys.executable is not available (e.g. inside pypy app-level)
+            # tests, then skip the rest of this test
+            return
+        # subprocess is not importable in pypy app-level tests
+        result = python_subprocess.run(mod, "mod.f()")
+        assert result.returncode == 0
+
```
Comment (antocuni) [inline test/test_hpyerr.py:684]: is it really needed to start a subprocess for this test? You could set `sys.unraisablehook` to something which records it has been invoked, then call `mod.f()` and check that the hook has been actually raised, unless I miss something

**Review by antocuni — COMMENTED** [review]
LGTM, apart the small comment about the test

File: test/test_hpyerr.py (hunk 1 above)
Comment (qunaibit) [reply test/test_hpyerr.py:699]: I didn't know about that.. nice suggestion. Thank you
I noticed that there is support for such routine in `test.support import catch_unraisable_exception` but this has only been added on the `3.8.12` release.. I added it manually until `hpy` drop support for older python releases

Comment (antocuni) [reply test/test_hpyerr.py:684]: ah ok, I didn't think about old pythons. +1 for subprocess then

Comment (qunaibit) [reply test/test_hpyerr.py:684]: approved for merge?

**Review by steve-s — APPROVED** [review]

**Pull request merged**